### PR TITLE
ResourceHealth - Swagger - AVAILABILITYSTATUSES API - 2020-05-01 Stable & Preview API versions

### PR DESCRIPTION
**Pull request by srrudray**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [x] Updating API in a stable or public preview version with Breaking Change Validation errors
- [x] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by srrudray** [comment]
Creating the PR to resolve the S360 - Swagger KPI Completeness 
KPI Link: 
https://vnext.s360.msftcloudes.com/blades/unifiedplatform?blade=AssignedTo:All~KPI:6dc4e485-2d1b-449c-a224-30b183391437~SLA:1~Forums:All~_loc:UnifiedPlatform&def=&peopleBasedNodes=ddove_team;prasank_team&tile=S360_ServiceId:ed05f3b7-8949-4698-9b38-49ab8abb5ee0~_loc:__key__UnifiedPlatform__6dc4e485-2d1b-449c-a224-30b183391437&global=4:ed05f3b7-8949-4698-9b38-49ab8abb5ee0

Short description: 
Recently we are trying to new API version 2020-05-01 and 2020-05-Preview for only one of our API (i.e AvailabilityStatuses API) 
Currently, Other APIs are Not using it.

I'm not sure, how to resolve some of the PR checks that are saying that why other API's are missing this API version

**Comment by akning-ms** [comment]
Hi @srrudray, You can refer to https://aka.ms/ci-fix (state in first comments)to fix the validation error. if still has issue to fix. you can refer to https://github.com/Azure/azure-rest-api-specs/pull/15502#issuecomment-892051703 to get further help.

And seems you are adding 2 new version,(stable & preview with same date part 2020-05-01)  can we know what different for them. 

and for our easy to review. best practice is (it can also be found in first comment) of this PR: 
Ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.

**Comment by srrudray** [comment]
> Hi @srrudray, You can refer to https://aka.ms/ci-fix (state in first comments)to fix the validation error. if still has issue to fix. you can refer to [#15502 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/15502#issuecomment-892051703) to get further help.
> 
> And seems you are adding 2 new version,(stable & preview with same date part 2020-05-01) can we know what different for them.
> 
> and for our easy to review. best practice is (it can also be found in first comment) of this PR:
> Ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.

1) The difference between 2 API version is 
Stable 2020-05-01 points to Production Traffic Manager
where as, the Preview 2020-05-01 points to Staging TraffiManager

This 2 versions are already existing in our ARM Manifest and are currently in use

**Comment by chiragg4u** [comment]
Please complete the checklist for ARM API Review to continue. Specifically following is required

Ensure you've reviewed following guidelines including ARM resource provider contract and REST guidelines. Estimated time (4 hours). This is required before you can request review from ARM API Review board.

**Comment by akning-ms** [comment]
> > Hi @srrudray, You can refer to https://aka.ms/ci-fix (state in first comments)to fix the validation error. if still has issue to fix. you can refer to [#15502 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/15502#issuecomment-892051703) to get further help.
> > And seems you are adding 2 new version,(stable & preview with same date part 2020-05-01) can we know what different for them.
> > and for our easy to review. best practice is (it can also be found in first comment) of this PR:
> > Ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.
> 
> 1. The difference between 2 API version is
>    Stable 2020-05-01 points to Production Traffic Manager
>    where as, the Preview 2020-05-01 points to Staging TraffiManager
> 
> This 2 versions are already existing in our ARM Manifest and are currently in use

I mean what difference for API definition? :) if it is same definition, only one version is ok. preview version is mainly used for your some new feature which is not stable which may change in the future. both preview or stable version can be used in production or staging.

**Comment by srrudray** [comment]
Breaking change exception Ticket - https://msazure.visualstudio.com/One/_workitems/edit/10564725

**Comment by srrudray** [comment]
Breaking change Exception ticket : https://msazure.visualstudio.com/One/_workitems/edit/10564725

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json
Comment (jorgecotillo) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json:278]: >localizedValue [](http://example.com/codeflow?start=13&length=14)

Out of curiosity, is this a text translated into a different language?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json
Comment (jorgecotillo) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json:538]: >"id" [](http://example.com/codeflow?start=8&length=4)

What is the motivation for using a reference to resource definition from common (ln 510) vs copying id, name, type and location? consider using trackedResource from definition.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json
Comment (jorgecotillo) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json:580]: >reasonType [](http://example.com/codeflow?start=13&length=10)

If this list is static, consider using enum

**Comment by jorgecotillo** [comment]
The PR looks good, left a few comments, please reply to them, since GH doesn't have approved with suggestions, I'll wait for the reply before signing off from ARM side

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json
Comment (srrudray) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json:278]: for now, we are just focused on English language

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json
Comment (srrudray) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json:580]: we want to keep it same as our other older API versions

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json
Comment (srrudray) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json:538]: This PR is to resolve the S360 Swagger KPI, due to the timeboud and delay, we just tried to follow the same format as our other current API versions and make it consistent everywhere

**Comment by jorgecotillo** [comment]
Signed off from ARM side

**Review by leniatgh — APPROVED** [review]

File: custom-words.txt
Comment (leni-msft) [inline custom-words.txt:1303]: typo? should be `occurred` ?

Comment (srrudray) [reply custom-words.txt]: reverted back

**Pull request merged**
